src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -198,13 +202,13 @@ private HashSet<CollectionName> ExecuteCodeAndCollectReferencedCollections(strin
         {
             _engine.SetValue("load", new ClrFunctionInstance(_engine, "load", LoadDocument));
 
-            _engine.Execute(Code);
+            _engine.ExecuteWithReset(Code);
```

Comment:
There is no limit to the number of steps it takes to _define_ maps / reduce

Reply:
`InitializeEngine` is wrapped with `DisableMaxStatements`

Reply:
So why use `ExecutewithReset` ?

Reply:
Thought it would be good to reset CallStack and TimeoutTicks before calling next Execute. This is what we are doing after each Execute in other places.